File: frontend/src/models/Review.ts
//elena
/**
 * review model representing user feedback for an event
 */
export interface Review {
    id: number;
    content: string;
    rating: number; //1-5
    createdAt: string | Date;
    eventId: number;
    employeeId: number;
    employeeName: string;
    employeeEmail: string;
}

export interface CreateReviewRequest {
    employeeId: number;
    content: string;
    rating: number; //1-5
}

File: backend/Models/Review.cs
// elena
namespace TodoApi.Models
{
    ///<summary>
    /// User review for an event 
    /// </summary>
    public class Review
    {
        public long Id { get; set; }
        public string Content { get; set; } = string.Empty;
        public int Rating { get; set; }
        public DateTime CreatedAt { get; set; }
        public long EventId { get; set; }
        public long EmployeeId { get; set; }
        public Event Event { get; set; } = null!;
        public string EmployeeName { get; set; } = string.Empty;
        public string EmployeeEmail { get; set; } = string.Empty;
    }
}
